DragonLoop/DragonLoopAPI/Managers/StopManager.cs

```diff
@@ -0,0 +1,81 @@
+﻿using DragonLoopModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DragonLoopAPI.Managers
+{
+    public class StopManager
+    {
+        /// <summary>
+        /// Backtracks route segments to find previous stops and check for buses on this route that
+        /// have last stopped there
+        /// </summary>
+        /// <param name="stop">The stop to find the next bus to arrive at</param>
+        /// <returns>The next bus to arrive at the given stop</returns>
+        public Bus GetNextBus(Stop stop)
+        {
+            Bus bus;
+            var routeSegment = stop.RouteSegment.PreviousRouteSegment;
+
+            //look through previous route segments until a stop is found that a bus has recently stopped at
+            while (routeSegment.FromStop == null || !TryGetBusAtStop(routeSegment.FromStop, out bus))
+            {
+                routeSegment = routeSegment.PreviousRouteSegment;
+            }
+
+            return bus;
+        }
+
+        /// <summary>
+        /// Get the first bus that has most recently stopped at the given stop
+        /// </summary>
+        /// <param name="stop">The stop to find buses at</param>
+        /// <param name="bus">The bus that first passed this stop</param>
+        /// <returns>If a bus was most recently at the given stop</returns>
+        private bool TryGetBusAtStop(Stop stop, out Bus bus)
```

Comment:
Aren't there multiple bus stop overlaps like the Powelton/Spring Garden Route and Dragon Route? How would you differentiate between buses that happen to be near the 33rd and market st stop?

Reply:
Those stops are multiple entries in our database so they're treated as separate stops. I don't love that, but I couldn't find a good way around it because of stuff like this method. If you have any ideas I'm open to any input about different ways to do our data modeling.